flowforge/input/Components.py

```diff
@@ -1765,11 +1810,26 @@ def _validate_hex_map(channel_map: List[List[str]]) -> None:
         num_rows = len(channel_map)
         assert num_rows > 0, "channel_map: must not be empty"
         assert num_rows % 2 == 1, f"channel_map: must have an odd number of rows, not {num_rows} rows"
-        n = (num_rows + 1) // 2
-        # expected lengths: [n, n+1, …, 2*n-1, 2*n-2, …, n]
-        ascending = list(range(n, 2 * n))
-        descending = list(range(2 * n - 2, n - 1, -1))
-        expected = ascending + descending
+        if orientation == "x":
+            n = (num_rows + 1) // 2
+            # For y-oriented hexagon (flat-top)
```

Comment:
should be `x-oriented` and the comment below should be `y-oriented`.  You have them swapped here.